src/main/java/com/blamejared/compat/tcomplement/highoven/MixRecipeManager.java

```diff
@@ -0,0 +1,130 @@
+package com.blamejared.compat.tcomplement.highoven;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import com.blamejared.compat.mantle.RecipeMatchIIngredient;
+import com.blamejared.mtlib.helpers.InputHelper;
+
+import crafttweaker.annotations.ModOnly;
+import crafttweaker.annotations.ZenRegister;
+import crafttweaker.api.item.IIngredient;
+import crafttweaker.api.liquid.ILiquidStack;
+import knightminer.tcomplement.library.events.TCompRegisterEvent;
+import knightminer.tcomplement.library.steelworks.MixAdditive;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import slimeknights.mantle.util.RecipeMatch;
+import stanhebben.zenscript.annotations.NotNull;
+import stanhebben.zenscript.annotations.ZenClass;
+import stanhebben.zenscript.annotations.ZenMethod;
+
+@ZenClass("mods.tcomplement.highoven.MixRecipeManager")
+@ZenRegister
+@ModOnly("tcomplement")
+public class MixRecipeManager {
+	private boolean init;
+	private FluidStack input, output;
+	private List<SimpleEntry<RecipeMatch, MixAdditive>> additives;
+	private Map<MixAdditive, List<RecipeMatch>> removedAdditives;
+
+	public MixRecipeManager(ILiquidStack output, ILiquidStack input) {
+		this.init = false;
+		this.input = InputHelper.toFluid(input);
+		this.output = InputHelper.toFluid(output);
+		this.additives = new LinkedList<>();
+		this.removedAdditives = new HashMap<>(MixAdditive.values().length, 1.0f);
+		for (MixAdditive type : MixAdditive.values()) {
+			this.removedAdditives.put(type, new LinkedList<>());
+		}
+	}
+
+	@ZenMethod
+	public MixRecipeManager addOxidizer(@NotNull IIngredient oxidizer, int consumeChance) {
+		this.additives.add(new SimpleEntry<>(new RecipeMatchIIngredient(oxidizer, consumeChance), MixAdditive.OXIDIZER));
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeManager addReducer(@NotNull IIngredient reducer, int consumeChance) {
+		this.additives.add(new SimpleEntry<>(new RecipeMatchIIngredient(reducer, consumeChance), MixAdditive.REDUCER));
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeManager addPurifier(@NotNull IIngredient purifier, int consumeChance) {
+		this.additives.add(new SimpleEntry<>(new RecipeMatchIIngredient(purifier, consumeChance), MixAdditive.PURIFIER));
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeManager removeOxidizer(IIngredient oxidizer) {
+		if (oxidizer != null && removedAdditives.get(MixAdditive.OXIDIZER) != null) {
+			removedAdditives.get(MixAdditive.OXIDIZER).add(new RecipeMatchIIngredient(oxidizer));
```

Comment:
Don't really need to use RecipeMatch for the removals since it never gets passed to the recipe, just use IIngredient directly for the match